### ISSUE TITLE
no-complex-conditionals: Prevent conditionals from being complex and potentially difficult to maintain in the future

### ISSUE THREAD

**Issue by Glavin001** [Glavin001/tslint-clean-code#3]
# Bad

```typescript
// Setup
const status: number = ...;
const isSkyBlue: boolean = ...;
const isRightDay: boolean = ...;

// Test
if (((status === 1 || status === 2 || status === 3) || isSkyBlue) && isRightDay) {
  doStuff();
}
```

# Good

```typescript
const shouldDoStuff = (((status === 1 || status === 2 || status === 3) || isSkyBlue) && isRightDay);
if (shouldDoStuff) {
  doStuff();
}

// OR (when in Class)

class StuffDoer {

  doTheThing() {
    if (this.shouldDoStuff()) {
      doStuff();
    }
  }

  shouldDoStuff() {
    return (((status === 1 || status === 2 || status === 3) || isSkyBlue) && isRightDay);
  }

}

```

**Comment by Glavin001** [comment]
Related to https://refactoring.guru/decompose-conditional
